service-print/src/main/java/org/oskari/print/loader/CommandLoadImageWMTS.java

```diff
@@ -201,6 +211,43 @@ private LayerCapabilitiesWMTS getLayerCapabilities() throws IllegalArgumentExcep
         throw new IllegalArgumentException("Could not find layer from Capabilities");
     }
 
+    private String getLayerAdditionalRequestParams() {
+        String[] layerUrlParams = layer.getOskariLayer().getUrl().split("\\?");
```

Comment:
OskariLayer has getParams(). We should have the apikey there instead of parsing it from the URL. Also IOHelper has bunch of methods parsing and constructing URLs.